### PR TITLE
docs(oauth): updates oauth config to show jsonpath options

### PR DESCRIPTION
**Pull request by PaulRMellor**
Documentation

Updates the OAUTH docs to show new support for JsonPath queries when configuring  `userNameClaim`, `fallbackUserNameClaim` + a few format updates

### Checklist

_Please go through this checklist and make sure all applicable tasks have been done_

- [ ] Write tests
- [ ] Make sure all tests pass
- [x] Update documentation
- [ ] Check RBAC rights for Kubernetes / OpenShift roles
- [ ] Try your changes from Pod inside your Kubernetes and OpenShift cluster, not just locally
- [ ] Reference relevant issue(s) and close them after merging
- [ ] Update CHANGELOG.md
- [ ] Supply screenshots for visual changes, such as Grafana dashboards

### CODE REVIEW

File: documentation/modules/oauth/con-oauth-authentication-broker.adoc [hunk 1]
```diff
@@ -93,6 +93,8 @@ NOTE: All communication with the authorization server should be performed using
 You can configure a certificate truststore as a Kubernetes Secret in your Strimzi project namespace, and use a `tlsTrustedCertificates` attribute to point to the Kubernetes Secret containing the truststore file.
 
 You might want to configure a `userNameClaim` to properly extract a username from the JWT token.
+If required, you can use a JsonPath expression like `['user.info'].['user.id']` to retrieve the username from a nested JSON structure.
```
[the remark this answers is not in the document]
Comment (PaulRMellor) [reply documentation/modules/oauth/con-oauth-authentication-broker.adoc:96]: I'm assuming the token. If so, we could change to "..._to retrieve the username from a token with a nested JSON structure._"

[the remark this answers is not in the document]
Comment (ppatierno) [reply documentation/modules/oauth/con-oauth-authentication-broker.adoc:96]: Ah that makes more sense, right! We are referring to the JWT token. Yeah I would add this "detail" here :-)

**Review by scholzj — APPROVED** [review]
LGTM. But hsould be reviewed by @mstruk as the SME.

**Review by ppatierno — APPROVED** [review]

File: documentation/modules/oauth/con-oauth-authentication-broker.adoc (hunk 1 above)
[the remark this answers is not in the document]
Comment (mstruk) [reply documentation/modules/oauth/con-oauth-authentication-broker.adoc]: We could be very specific here and say 'JWT token payload'. JWT token has three section with 'payload' section containing a JSON object.

[the remark this answers is not in the document]
Comment (mstruk) [reply documentation/modules/oauth/con-oauth-authentication-broker.adoc]: But then, we also support opaque tokens and introspection endpoint, and user endpoint where the JSON object is simply a response from the endpoint. So saying 'JWT token payload' is too specific.

[the remark this answers is not in the document]
Comment (mstruk) [reply documentation/modules/oauth/con-oauth-authentication-broker.adoc]: Saying 'from a token with a nested JSON structure' seems to imply (to me) that there are situations where we're not dealing with a JSON structure which is not the case. Maybe we could simply say: 'from nested attributes'.

[the remark this answers is not in the document]
Comment (PaulRMellor) [reply documentation/modules/oauth/con-oauth-authentication-broker.adoc]: If we reverse it, I think we make it clear without that implication: _If required, you can use a JsonPath expression like `"['user.info'].['user.id']"` to retrieve the username from nested JSON attributes within a token._

**Review by mstruk — APPROVED** [review]

**Pull request merged**